admin/src/main/scala/io/buoyant/admin/Admin.scala

```diff
@@ -1,65 +1,112 @@
 package io.buoyant.admin
 
-import com.twitter.app.{App => TApp}
 import com.twitter.finagle._
 import com.twitter.finagle.http.{HttpMuxer, Request, Response}
-import com.twitter.finagle.stats.MetricsStatsReceiver
+import com.twitter.finagle.stats.{MetricsStatsReceiver, NullStatsReceiver}
+import com.twitter.finagle.tracing.NullTracer
 import com.twitter.logging.Logger
 import com.twitter.server.Admin.Path
-import com.twitter.server.handler.{SummaryHandler => TSummaryHandler, _}
-import com.twitter.server.view.{IndexView, TextBlockView}
+import com.twitter.server.handler.{SummaryHandler => _, _}
+import com.twitter.server.view.{IndexView, TextBlockView, NotFoundView}
+import com.twitter.util.Monitor
+import java.net.SocketAddress
 
-class Admin(app: TApp) {
+object Admin {
+  val label = "adminhttp"
+  private val log = Logger.get(label)
 
-  private[this] val log = Logger()
+  type Handler = Service[Request, Response]
+  type Handlers = Seq[(String, Handler)]
 
-  private[this] def twitterServerRoutes: Seq[(String, Service[Request, Response])] = Seq(
-    "/admin" -> new TSummaryHandler,
-    "/admin/server_info" -> (new TextBlockView andThen new ServerInfoHandler(app)),
+  /**
+   * A type for modules that expose admin handlers.
+   */
+  trait WithHandlers {
+    def adminHandlers: Handlers
+  }
+
+  def getHandlers(obj: Any): Handlers =
+    obj match {
+      case wh: WithHandlers => wh.adminHandlers
+      case _ => Nil
+    }
+
+  def collectHandlers(objs: Seq[Any]): Handlers =
+    objs.flatMap(getHandlers(_))
```

Comment:
not required, but it signals (to me) that there's a function allocation here.  Got this convention from finagle.